node/consensus/state/header.go

```diff
@@ -42,6 +46,20 @@ func (h *Header) Deserialize(bytes []byte) error {
 		pos += availableBlockSerializedSize
 		h.AvailableBlocks = append(h.AvailableBlocks, ab)
 	}
+	availableCommonBlockCount := int(binary.BigEndian.Uint32(bytes[pos : pos+4]))
+	pos += 4
+	h.AvailableCommonBlocks = nil
```

Comment:
You can make a slice of size `availableCommonBlockCount`, and assign to it with `i`